scripts/gulp/frontend-shared.js

```diff
@@ -25,22 +27,29 @@ const buildFrontendSharedJs = () => {
   );
 };
 
-const linkFrontendShared = done => {
-  // Setup a symlink from the client to the frontend-shared package.
-  cp.spawn('yarn', ['link'], {
-    env: process.env,
-    cwd: './frontend-shared',
-  }).on('exit', () => {
-    cp.spawn('yarn', ['link', '@hypothesis/frontend-shared'], {
-      env: process.env,
-      stdio: 'inherit',
-    }).on('exit', () => {
-      done();
-    });
-  });
+const buildFrontendSharedTypes = async () => {
+  // nb. If the options get significantly more complex, they should be moved to
+  // a `tsconfig.json` file.
+  await run('node_modules/.bin/tsc', [
```

Comment:
Would this be more modern or a dependency we don't want to force?

`await run('npx', [
    'tsc', ....`

Reply:
Using `npx` might work for this purpose. It is bundled with npm >= 5.2 apparently, so we can probably rely on it. I'm not sure how much overhead it has vs. just invoking the script directly.